stew/results.nim

```diff
@@ -953,25 +953,52 @@ template isNone*(o: Opt): bool =
 # Syntactic convenience
 
 template `?`*[T, E](self: Result[T, E]): auto =
+  mixin assignResult
```

Comment:
Since this is a very low-level integration mechanism between advanced Nim libraries, I suggest relying on the Nim `backticks` feature and use an identifier that's extremely unlike to appear in user code (e.g. `#assignResult`). It's also worth re-reading the discussion we had in the Chronos tracker:

https://github.com/status-im/nim-chronos/pull/297#discussion_r927499892

Reply:
`#assignResult` doesn't compile - `assignResult?` is fine however, switched